graphql_compiler/tests/test_compiler.py

```diff
@@ -4171,6 +4170,36 @@ def test_fold_on_output_variable(self):
               Animal___1.name AS `animal_name`,
               [x IN collected_Animal__out_Animal_ParentOf___1 | x.name] AS `child_names_list`
         '''
+        expected_mssql = SKIP_TEST
+
+        check_test_data(self, test_data, expected_match, expected_gremlin, expected_mssql,
+                        expected_cypher, expected_postgresql)
+
+    def test_fold_on_foreign_key(self):
+        test_data = test_input_data.fold_on_foreign_key()
+
+        expected_postgresql = '''
+            SELECT
+                "Animal_1".name AS animal_name,
+                coalesce(folded_subquery_1.fold_output_name, ARRAY[]::VARCHAR[]) AS homes_list
+            FROM
+                schema_1."Animal" AS "Animal_1"
+            LEFT OUTER JOIN (
+                SELECT
+                    "Animal_2".uuid AS uuid,
+                    array_agg("Location_1".name) AS fold_output_name
+                FROM schema_1."Animal" AS "Animal_2"
+                JOIN schema_1."Location" AS "Location_1" ON "Animal_2".lives_in = "Location_1".uuid
+                GROUP BY
+                    "Animal_2".uuid
+            ) AS folded_subquery_1
+            ON "Animal_1".uuid = folded_subquery_1.uuid
+        '''
+
+        expected_mssql = SKIP_TEST
+        expected_match = SKIP_TEST
+        expected_gremlin = SKIP_TEST
+        expected_cypher = SKIP_TEST
```

Comment:
Should these be implemented eventually? If so, put a TODO. If not, put a comment about why not.